subsys/testsuite/ztest/src/ztest_new.c

```diff
@@ -495,6 +500,22 @@ struct ztest_unit_test *ztest_get_next_test(const char *suite, struct ztest_unit
 	return NULL;
 }
 
+bool z_ztest_should_test_run(const char *suite, const char *test)
+{
+	bool run_test = false;
+	char suite_test_pair[128];
+	char suite_test_wild[128];
+
+	sprintf(suite_test_pair, "%s::%s", suite, test);
+	sprintf(suite_test_wild, "%s::*", suite);
+
+	run_test = (test_args == NULL ||
+		    strstr(test_args, suite_test_pair) != NULL ||
+		    strstr(test_args, suite_test_wild) != NULL);
```

Comment:
I don't think we can safely assume that `suite::test` will be under 128 chars as there's no documentation that specifies that. I think we should be implementing our own loop that looks for these in test_args. Should be a simple recursive call (which is fine since this is only used on ARCH_POSIX)

Reply:
Since moving this to its own file for ARCH_POSIX, I'm more comfortable with using malloc to allocate necessary space to generate the formatted strings.